aten/src/ATen/native/quantized/cuda/affine_quantizer.cu

```diff
@@ -1,15 +1,29 @@
+#include <ATen/ATen.h>
+#include <ATen/NativeFunctions.h>
 #include <ATen/native/TensorIterator.h>
 #include <ATen/native/quantized/affine_quantizer.h>
 #include <math.h>
 #include <ATen/cuda/CUDAApplyUtils.cuh>
 #include <ATen/native/cuda/Loops.cuh>
-#include <ATen/ATen.h>
-#include <ATen/NativeFunctions.h>
 
 namespace at {
 namespace native {
 namespace {
 
+template <typename T>
+void check_zero_points_cuda(
+    const std::string& fn_name,
+    const Tensor& zero_points) {
+  constexpr int64_t qmin = std::numeric_limits<T>::min();
+  constexpr int64_t qmax = std::numeric_limits<T>::max();
+  auto zp_within_upper = at::any(at::gt(zero_points, qmax)).item().equal(false);
+  auto zp_within_lower = at::any(at::lt(zero_points, qmin)).item().equal(false);
+  TORCH_CHECK(
+    zp_within_upper and zp_within_lower,
+    fn_name,
+    " zero_point is out of range.");
```

Comment:
Maybe adding information of what is out of range would be useful (lower/upper)

Reply:
I'm not sure, I was under the impression that anything external would go through the dispatcher

Reply:
good idea, fixed